### DISCUSSION TITLE
Wonder if 24 bit fixed point is possible

### DISCUSSION THREAD

**Discussion by VinzyNg** [ExistentialAudio/BlackHole#446] [Feature Requests]
I want to avoid any unnecessary dither or truncation on 32bit float to 24bit fixed point conversion, so I wonder if 24bit fixed point direct routing is possible. 

thanks!

**Comment by devinroth** [comment]
There is absolutely no loss in resolution from 32bit float to 24bit int. 32bit float is capable of encoding 24bit int without any loss, dither or truncation. Converting from Int24 to Float32 to Int24 will result in the exact same values.

Completely unnecessary. On top of that the Hardware Abstraction Layer handles all audio in 32bit float regardless so even if you were to connect to the driver with 24 bit int the HAL would still convert it to 32bit float.

The downside of 32bit float is it takes up more space, and on older machines and less powerful chips it’s a little more expensive on the CPU. Virtual no difference on new machines.

The benefit is that there is no clipping. You can push the limit beyond 1.0 with only loss in resolution instead of clipping.

In short Float32 has a single value for every 24bit int value between the range of -1.0 to 1.0.

Devin Roth
